app/controllers/IndexController.scala

```diff
@@ -42,25 +42,33 @@ class IndexController @Inject() (
 
   def index(departureId: String): Action[AnyContent] = actions.getData(departureId).async {
     implicit request =>
-      service.getLRN(departureId) flatMap {
-        lrn =>
-          sessionRepository
-            .set(
-              request.userAnswers.getOrElse(
-                UserAnswers(
-                  departureId,
-                  request.eoriNumber,
-                  lrn.value,
-                  JsObject.empty,
-                  timeMachine.now()
+      service.getDepartureData(departureId).flatMap {
+        case Some(departureData) =>
+          service.getLRN(departureId) flatMap {
```

Comment:
We can retrieve the LRN from the MessageMetaData so we can remove this call and add it to the reads for MessageMetaData:
https://developer.service.hmrc.gov.uk/api-documentation/docs/api/service/common-transit-convention-traders/2.0/oas/page#tag/Departures/operation/getCachedDepartureForDepartureID